contracts/modules/cover/Cover.sol

```diff
@@ -327,9 +332,17 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
             lastCoverSegment.start,
             lastCoverSegment.period,
             originalPoolAllocations[i].coverAmountInNXM,
-            originalPoolAllocations[i].premiumInNXM / REWARD_DENOMINATOR
+            originalPoolAllocations[i].premiumInNXM,
+            lastCoverSegment.globalRewardsRatio
           );
+
           originalPoolAllocations[i].premiumInNXM *= (lastCoverSegment.period - remainingPeriod) / lastCoverSegment.period;
+
+          // compute NXM rewards deallocated
+          uint deallocatedRewardsInNXM = originalPoolAllocations[i].premiumInNXM
+          * remainingPeriod / lastCoverSegment.period
+          * lastCoverSegment.globalRewardsRatio / REWARD_DENOMINATOR;
+          tokenController().burnPooledStakingNXMRewards(deallocatedRewardsInNXM, originalPoolAllocations[i].poolId);
```

Comment:
Let's create 2 arrays, one with nxm amounts, one with pool ids, and pass them in 1 call to TokenController

Reply:
discussed off-band. we can keep it with 1 param instead of array, you save around 100 gas and it's a rare case that you have multiple allocations.